nifi-commons/nifi-property-utils/src/main/java/org/apache/nifi/properties/BootstrapProperties.java

```diff
@@ -79,8 +107,53 @@ private String getPropertyKey(final String subKey) {
         return Optional.ofNullable(getProperty(getPropertyKey(BOOTSTRAP_SENSITIVE_KEY)));
     }
 
+    /**
+     * Returns the location of the HashiCorp Vault sensitive property provider
+     * @return The location of the HashiCorp Vault sensitive property provider
+     */
+    public Optional<String> getHashiCorpVaultPropertiesFile() {
+        return Optional.ofNullable(getProperty(getPropertyKey(BOOTSTRAP_SPP_HASHICORP_VAULT_PROPERTIES)));
+    }
+
     @Override
     public String toString() {
         return String.format("Bootstrap properties [%s] with prefix [%s]", configFilePath, propertyPrefix);
     }
+
+    /**
+     * An empty instance of BootstrapProperties.
+     */
+    public static final BootstrapProperties EMPTY = new BootstrapProperties("", new Properties(), Paths.get("conf/bootstrap.conf")) {
+        @Override
+        public Set<String> getPropertyKeys() {
+            return null;
+        }
+
+        @Override
+        public String getProperty(String key) {
+            return null;
+        }
+
+        @Override
+        public String getProperty(String key, String defaultValue) {
+            return null;
+        }
+
+        @Override
+        public int size() {
+            return 0;
+        }
+    };
+
+    private static boolean isBlank(final String string) {
+        return (string == null) || string.isEmpty() || string.trim().isEmpty();
+    }
+
+    /**
+     * Returns the property prefix for these properties (e.g., 'nifi').
+     * @return The property prefix
+     */
+    public String getPropertyPrefix() {
```

Comment:
Should this method be public?

Reply:
Actually this was part of something I abandoned, so I'll just remove it.